app/models/suspicious_behavior.rb

```diff
@@ -0,0 +1,64 @@
+# == Schema Information
+#
+# Table name: suspicious_behaviors
+#
+#  id                  :bigint           not null, primary key
+#  behavior_type       :integer          not null
+#  flagged_at          :datetime         not null
+#  metadata            :text(4294967295)
+#  resolved_at         :datetime
+#  status              :integer          default("pending"), not null
+#  created_at          :datetime         not null
+#  updated_at          :datetime         not null
+#  collection_id       :integer
+#  page_id             :integer
+#  resolved_by_user_id :integer
+#  user_id             :integer          not null
+#
+# Indexes
+#
+#  index_suspicious_behaviors_on_collection_id        (collection_id)
+#  index_suspicious_behaviors_on_page_id              (page_id)
+#  index_suspicious_behaviors_on_resolved_by_user_id  (resolved_by_user_id)
+#  index_suspicious_behaviors_on_user_id              (user_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (collection_id => collections.id) ON DELETE => nullify
+#  fk_rails_...  (page_id => pages.id) ON DELETE => nullify
+#  fk_rails_...  (resolved_by_user_id => users.id) ON DELETE => nullify
+#  fk_rails_...  (user_id => users.id) ON DELETE => cascade
+#
+class SuspiciousBehavior < ApplicationRecord
+  belongs_to :user
+  belongs_to :page, optional: true
+  belongs_to :collection, optional: true
+  belongs_to :resolved_by_user, optional: true, class_name: 'User'
+
+  enum :behavior_type, {
```

Comment:
In the `Page` model, we were able to use string values for our status `enum`.  This gives us the benefit of `enum` behavior while still letting raw DB access look readable.  Could we do that here instead of using integers?